backend/src/main/java/wooteco/retrospective/infrastructure/dao/pair/PairDaoImpl.java

```diff
@@ -0,0 +1,142 @@
+package wooteco.retrospective.infrastructure.dao.pair;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+import wooteco.retrospective.domain.attendance.Attendance;
+import wooteco.retrospective.domain.attendance.Time;
+import wooteco.retrospective.domain.dao.PairDao;
+import wooteco.retrospective.domain.member.Member;
+import wooteco.retrospective.domain.pair.Pair;
+import wooteco.retrospective.domain.pair.Pairs;
+
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toList;
+
+@Repository
+public class PairDaoImpl implements PairDao {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    public PairDaoImpl(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Transactional
+    public Pairs insert(Pairs pairs) {
+        String sql = "INSERT INTO PAIR (group_id, attendance_id) VALUES (?, ?)";
+
+        pairs.getPairs().forEach(
+                pair -> jdbcTemplate.batchUpdate(sql, getValuesForEachPair(pair))
+        );
+
+        return pairs;
+    }
+
+    private List<Object[]> getValuesForEachPair(Pair pair) {
+        return pair.getAttendances().stream().map(
+                attendance -> new Object[]{pair.getGroupId(), attendance.getId()}
+        ).collect(toList());
+    }
+
+    private List<Pair> membersToPairs(List<AttendanceWithGroupId> members) {
+        Map<Long, List<AttendanceWithGroupId>> membersWithGroupId = members.stream()
+                .collect(groupingBy(AttendanceWithGroupId::getGroupId));
+
+        return membersWithGroupId.entrySet().stream()
+                .map(this::toPair)
+                .collect(toList());
+    }
+
+    private Pair toPair(Map.Entry<Long, List<AttendanceWithGroupId>> entry) {
+        return new Pair(
+                entry.getKey(), entry.getValue().stream()
+                .map(AttendanceWithGroupId::getAttendance)
+                .collect(toList())
+        );
+    }
+
+    public Optional<Pairs> findByDateAndTime(LocalDate date, Time time) {
+        List<AttendanceWithGroupId> members = jdbcTemplate.query(
+                getFindByDateAndTimeSql(),
+                rowMapperForFindByDateAndTime(time),
+                date,
+                time.getTime()
+        );
+
+        if (members.isEmpty()) {
+            return Optional.empty();
+        }
+
+        List<Pair> pairs = membersToPairs(members);
+
+        return Optional.of(Pairs.from(pairs));
```

Comment:
메서드 명이 잘 지어져있어서 이해하기 쉬운데 이렇게 inline 하는건 어떤가요~~~?
```suggestion 
        return Optional.of(Pairs.from(membersToPairs(members)));
```

Reply:
좋아요~